Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Reviews/CQRS/Handlers/Commands/UpdateReviewCommandHandler.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Threading.Tasks;
+using AutoMapper;
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Application.Exceptions;
+using BlogApp.Application.Features.Blog.DTOs;
+using BlogApp.Application.Features.Reviews.CQRS.Commands;
+using BlogApp.Application.Features.Reviews.DTOs;
+using BlogApp.Application.Features.Reviews.DTOs.Validators;
+using BlogApp.Application.Responses;
+using BlogApp.Domain;
+using MediatR;
+
+namespace BlogApp.Application.Features.Reviews.CQRS.Handlers.Commands
+{
+    public class UpdateReviewCommandHandler : IRequestHandler<UpdateReviewCommand, Result<ReviewDto>>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public UpdateReviewCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<Result<ReviewDto>> Handle(UpdateReviewCommand request, CancellationToken cancellationToken)
+        {
+            var response = new Result<ReviewDto>();
+
+            if (request.reviewIsApprovedDto != null)
+            {
+                var validator = new UpdateIsResolvedValidator(_unitOfWork);
+                var validationResult = await validator.ValidateAsync(request.reviewIsApprovedDto);
+
+                if (validationResult.IsValid == true)
+                {
+                    var review = await _unitOfWork.ReviewRepository.Get(request.reviewIsApprovedDto.Id);
```

Comment:
If you have handled to check wether the review with this Id exist or not at your validator this is okay...but if you haven't added that you should check if the review is null or not(if it exists or not).